patches/minecraft/net/minecraft/command/impl/SpreadPlayersCommand.java.patch

```diff
@@ -0,0 +1,12 @@
+--- a/net/minecraft/command/impl/SpreadPlayersCommand.java
++++ b/net/minecraft/command/impl/SpreadPlayersCommand.java
+@@ -162,7 +_,8 @@
+             spreadplayerscommand$position = p_241072_2_[i++];
+          }
+ 
+-         entity.func_223102_j((double)MathHelper.func_76128_c(spreadplayerscommand$position.field_198713_a) + 0.5D, (double)spreadplayerscommand$position.func_198710_a(p_241072_1_, p_241072_3_), (double)MathHelper.func_76128_c(spreadplayerscommand$position.field_198714_b) + 0.5D);
++         net.minecraftforge.event.entity.living.EntityTeleportEvent.SpreadPlayersCommand event = net.minecraftforge.event.ForgeEventFactory.onEntityTeleportSpreadPlayersCommand(entity, (double)MathHelper.func_76128_c(spreadplayerscommand$position.field_198713_a) + 0.5D, (double)spreadplayerscommand$position.func_198710_a(p_241072_1_, p_241072_3_), (double)MathHelper.func_76128_c(spreadplayerscommand$position.field_198714_b) + 0.5D);
++         if (!event.isCanceled()) entity.func_223102_j(event.getTargetX(), event.getTargetY(), event.getTargetZ());
```

Comment:
Else condition for the event here? Shouldn't this perform an early exit or continue?

Reply:
The patch does not fully show the scope of this method execution. The code in question is actually executed in a loop for all entities selected by the selector or w/e in `/spreadplayers`. Exiting early would cause all entities after the one that is canceled from teleportation to also be canceled, which would be an unintended side effect. This way, entities can be selectively blocked from teleporting instead of all of them.